app/assets/stylesheets/components/_navigation.scss

```diff
@@ -1,30 +1,100 @@
+body {
+  margin: 0;
+}
+
 .c-navigation {
+  height: 50px;
   background-color: colour(dark-grey);
-  height: 40px;
-
-  &__links {
+  a {
+    text-decoration: none;
+    cursor: pointer;
+  }
+  &__checkbox {
+    visibility: hidden;
+    &:checked~.c-navigation__menu {
+      left: 0%;
+    }
+  }
+  &__menu {
+    overflow: auto;
+    width: 100vw;
+    height: 100%;
+    left: 110%;
+    background-color: colour(dark-grey);
+    position: absolute;
+    z-index: 999;
+    transition: left 0.4s;
+    margin: 0;
+    top: 80px;
     list-style-type: none;
     margin-left: auto;
     margin-right: auto;
-    vertical-align: middle;
     a {
-      color: colour(white);
-      vertical-align: middle;
-      line-height: 40px;
+      color: colour(heading-orange);
+      cursor: pointer;
+      text-decoration: none;
+
+      &:hover {
+        color: colour(border-blue);
+        transition: ease-in-out 0.3s all;
+      }
     }
     li {
-      display: inline-block;
-      vertical-align: middle;
+      display: block;
+      padding-top: $base-spacing-unit;
     }
     ul {
       margin-bottom: 0;
       margin-top: 0;
-      vertical-align: middle;
       text-align:center;
     }
+    &_list {
+      list-style: none;
+      padding-left: 3rem;
+    }
+    &_button {
+      display: block;
+      background: url("https://svgshare.com/i/7X_.svg") no-repeat center center;
+      text-indent: -200%;
+      width: 37px;
+      height: 37px;
+      float: right;
+      margin-top: 0.5rem;
+      color: black;
+      margin-right: 2rem;
+    }
   }
-  &__link {
-    padding: 0 40px;
-    text-decoration: none;
+}
+
+@media (min-width: 600px) {
```

Comment:
This can be nested above, and then you don't need to define `.c-navigation` again